src/Stripe.net/Entities/Checkout/Sessions/SessionAfterExpirationRecovery.cs

```diff
@@ -0,0 +1,39 @@
+// File generated from our OpenAPI spec
+namespace Stripe.Checkout
+{
+    using System;
+    using Newtonsoft.Json;
+    using Stripe.Infrastructure;
+
+    public class SessionAfterExpirationRecovery : StripeEntity<SessionAfterExpirationRecovery>
+    {
+        /// <summary>
+        /// Enables user redeemable promotion codes on the recovered Checkout Sessions. Defaults to
+        /// <c>false</c>.
+        /// </summary>
+        [JsonProperty("allow_promotion_codes")]
+        public bool? AllowPromotionCodes { get; set; }
+
+        /// <summary>
+        /// If <c>true</c>, a recovery url will be generated to recover this Checkout Session if it
+        /// expires before a transaction is completed. It will be attached to the Checkout Session
+        /// object upon expiration.
+        /// </summary>
+        [JsonProperty("enabled")]
+        public bool? Enabled { get; set; }
+
+        /// <summary>
+        /// The timestamp at which the recovery URL will expire.
+        /// </summary>
+        [JsonProperty("expires_at")]
+        [JsonConverter(typeof(UnixDateTimeConverter))]
+        public DateTime? ExpiresAt { get; set; }
```

Comment:
Why is this nullable? Shouldn't it be either always set or the entire hash is null?

Reply:
Good catch - from the checkout team:
> `after_expiration.recovery.enabled` should be required - `after_expiration.recovery.allow_promotion_codes` is optional on the write side with a default of false - so the resource [should] have that field as required as well. `after_expiration.recovery.url` and `after_expiration.recovery.expires_at` will be null until the Checkout Session expires, so those are correct